commands/webhook.go

```diff
@@ -0,0 +1,472 @@
+	Long:    "Modify existing outgoing webhook by changing its title, description, channel, icon, url, content-type, and triggers",
+	Example: `  webhook modify-outgoing [webhookId] --channel [channelId] --display-name [displayName] --description "New webhook description" --icon http://localhost:8000/my-slash-handler-bot-icon.png --url http://localhost:8000/my-webhook-handler --content-type "application/json" --trigger-word test --trigger-when start`,
+	RunE:    withClient(modifyOutgoingWebhookCmdF),
+}
+
+var WebhookDeleteCmd = &cobra.Command{
+	Use:     "delete",
+	Short:   "Delete webhooks",
+	Long:    "Delete webhook with given id",
+	Example: "  webhook delete [webhookID]",
+	RunE:    withClient(deleteWebhookCmdF),
+}
+
+func listWebhookCmdF(c client.Client, command *cobra.Command, args []string) error {
+	printer.SetSingle(true)
+
+	var teams []*model.Team
+	if len(args) < 1 {
+		var response *model.Response
+		// If no team is specified, list all teams
+		teams, response = c.GetAllTeams("", 0, 100000000)
+		if response.Error != nil {
+			return response.Error
+		}
+	} else {
+		teams = getTeamsFromTeamArgs(c, args)
+	}
+
+	for i, team := range teams {
+		if team == nil {
+			printer.PrintError("Unable to find team '" + args[i] + "'")
+			continue
+		}
+
+		// Fetch all hooks with a very large limit so we get them all.
+		incomingResult := make(chan StoreResult, 1)
+		go func() {
+			incomingHooks, response := c.GetIncomingWebhooksForTeam(team.Id, 0, 100000000, "")
+			incomingResult <- StoreResult{Data: incomingHooks, Err: response.Error}
+			close(incomingResult)
+		}()
+		outgoingResult := make(chan StoreResult, 1)
+		go func() {
+			outgoingHooks, response := c.GetOutgoingWebhooksForTeam(team.Id, 0, 100000000, "")
+			outgoingResult <- StoreResult{Data: outgoingHooks, Err: response.Error}
+			close(outgoingResult)
+		}()
+
+		if result := <-incomingResult; result.Err == nil {
+			printer.Print(fmt.Sprintf("Incoming webhooks for %s (%s):", team.DisplayName, team.Name))
+			hooks := result.Data.([]*model.IncomingWebhook)
+			for _, hook := range hooks {
+				printer.Print("\t" + hook.DisplayName + " (" + hook.Id + ")")
+			}
+		} else {
+			printer.PrintError("Unable to list incoming webhooks for '" + args[i] + "'")
+		}
+
+		if result := <-outgoingResult; result.Err == nil {
+			hooks := result.Data.([]*model.OutgoingWebhook)
+			printer.Print(fmt.Sprintf("Outgoing webhooks for %s (%s):", team.DisplayName, team.Name))
+			for _, hook := range hooks {
+				printer.Print("\t" + hook.DisplayName + " (" + hook.Id + ")")
+			}
+		} else {
+			printer.PrintError("Unable to list outgoing webhooks for '" + args[i] + "'")
+		}
+	}
+	return nil
+}
+
+func createIncomingWebhookCmdF(c client.Client, command *cobra.Command, args []string) error {
+	printer.SetSingle(true)
+
+	channelArg, errChannel := command.Flags().GetString("channel")
+	if errChannel != nil || channelArg == "" {
+		return errors.New("Channel is required")
+	}
```

Comment:
This can be replaced adding the following right after declaring the flag:

```go
WebhookCreateIncomingCmd.MarkFlagRequired("channel")
```

and then, when reading the flag:

```go
channelArg, _ := command.Flags().GetString("channel")
```